inert.js

```diff
@@ -69,13 +69,44 @@ class InertRoot {
     // Make all focusable elements in the subtree unfocusable and add them to _managedNodes
     this._makeSubtreeUnfocusable(this._rootElement);
 
+    const boundOnMutation = this._onMutation.bind(this);
     // Watch for:
     // - any additions in the subtree: make them unfocusable too
     // - any removals from the subtree: remove them from this inert root's managed nodes
     // - attribute changes: if `tabindex` is added, or removed from an intrinsically focusable element,
     //   make that node a managed node.
-    this._observer = new MutationObserver(this._onMutation.bind(this));
+    this._observer = new MutationObserver(boundOnMutation);
     this._observer.observe(this._rootElement, { attributes: true, childList: true, subtree: true });
+
+    // Watch for:
+    // - any additions in the shadowRoot subtree: make them unfocusable too
+    // - any removals from the shadowRoot subtree: remove them from this inert root's managed nodes
+    const rootObserver = new MutationObserver(boundOnMutation);
+    const shadowRoot = this._rootElement.shadowRoot || this._rootElement.webkitShadowRoot;
+    if (shadowRoot) {
+      rootObserver.observe(shadowRoot, { childList: true, subtree: true });
```

Comment:
@alice actually here we shoudn't observe attributes, as `MutationObserver` will observe the ones of the target. In this case, the target is the shadowRoot which won't get any attribute change. `this._observer` will take care of that.